nvidia.go

```diff
@@ -40,13 +40,18 @@ func getDevices() []*pluginapi.Device {
 	for i := uint(0); i < n; i++ {
 		d, err := nvml.NewDeviceLite(i)
 		check(err)
+		var cpu_affinity int64 = 0
+		if d.CPUAffinity != nil {
+			cpu_affinity = int64(*(d.CPUAffinity))
```

Comment:
If d.CPUAffinity is not set, then we shouldn’t force it to 0. Instead, we should just not set the Topology field of the device at all.

When is it the case that d.CPUAffinity is not set, such that this occurs?

Reply:
I got the same problem.

Since `NewDeviceLite` does not get`CPUAffinity`, `CPUAffinity` will always be nil (maybe).
https://github.com/NVIDIA/gpu-monitoring-tools/blob/0474d08c7a070ac97e9723d8f18f9cfaa64d5918/bindings/go/nvml/nvml.go#L395

If`NewDevice` is used, `CPUAffinity` may be obtained.
https://github.com/NVIDIA/gpu-monitoring-tools/blob/0474d08c7a070ac97e9723d8f18f9cfaa64d5918/bindings/go/nvml/nvml.go#L371

Reply:
@klueska So should NewDeviceLite() be replaced by NewDevice()，which always sets CPUAffinity in numaNode() ?

Reply:
I think it may be more complicated than that, because `NewDevice()` currently always sets CPUAffinity to "something" (e.g. 0) even if the underlying host doesn't have any NUMA info for it (i.e. the system returns -1).  I think we will need to change NVML to change the way it reports things, in addition to changing the plugin.  I plan to look into this later today.

Reply:
Once this merges https://gitlab.com/nvidia/container-toolkit/gpu-monitoring-tools/merge_requests/4, we can update the vendoring for the NVML bindings and update this patch to selectively set `Topology` based on if `d.CPUAffinity` is `nil` or not.

Reply:
OK. That change has merged. You will need to update the vendoring to pull it in and then update the logic here to only set `Topology` if `d.CPUAffinity != nil`.